### PR TITLE
CMS website status 500 page

### LINKED ISSUE #3742: Create static CMS status page when website is down
### Summary

**What we're after:**
_To better handle messaging to users while our website is down, we need to create a static CMS page that can be served up to inform users._

### Related issues

See this issue for related design mock ups

- #3694 

### Completion criteria

- [x] Within the CMS, create an status page when the website is down
- [ ] ~The page should be accessible even when the Wagtail DB is down~ The site will fail if the Wagtail DB is down. So there is no way to show any error messaging if this happens.
- [x] Add emergency page to the incident response documentation so that the team knows when to use it 

### Tech steps or considerations

- [x] Create a separate emergency status template
- [x] Add a feature flag for this template so that it can be turned on as needed
- [ ] ~Ensure that the page does not fail when the Wagtail DB is down~ The site will fail if the Wagtail DB is down. So there is no way to show any error messaging if this happens.

### PR DESCRIPTION
**Pull request by patphongs**
## Summary

- Resolves #3742 
_Introduces new 500 error page template under a feature flag. This alternative 500 error message can be turned on with a redeploy._

## Impacted areas of the application

List general components of the application that this PR will affect:

-  500 error page messaging

### Normal/Default 500 error message page
Screen Shot 2020-05-12 at 11 38 53 AM

### Alternative 500 error message page:
Screen Shot 2020-05-12 at 11 39 29 AM

## How to test

- Pull down the branch
- Turn off DEBUG mode inside of the dev.py file. Change to `DEBUG = False`. Also add `ALLOWED_HOSTS = ['*']`. These will be temporary changes to test out the 500 error messaging.
- `npm run build-js`
- Build assets using collectstatic so that scripts and styles will work on site: `./manage.py collectstatic --noinput -v 0`
- Set the feature flag `export FEC_FEATURE_WEBSITE_STATUS=true`. This will turn on the alternative 500 error messaging (see screenshot above for what that looks like).
- Pick a jinja template or a HTML template (our templates all use jinja as the framework even though the extension is .html). 
- Modify that chosen jinja template's extends to reference an invalid file name. Example: `{% extends 'layouts/invalid_name.jinja' %}`
- Run the server `./manage.py runserver`
- Go to the page that uses that modified jinja template. You should see the alternative 500 error message. 
- Test turning off the alternative 500 error message by 
    - Turn off your CMS server
    - Unset the flag: `unset FEC_FEATURE_WEBSITE_STATUS`
- Run the server `./manage.py runserver`
- Refresh the page with the modified jinja template and you should now see the normal/default 500 error message as seen in the screenshot above.

____

### CODE REVIEW

**Comment by johnnyporkchops** [comment]
@patphongs Related to an issue you pointed out with the environment not showing on the env-specific banners for 500 errors: https://github.com/fecgov/fec-cms/issues/3708.
We could pass it into this view. But this could also just be a done for as a followup issue. 

``` 
# Custom handle 500 error for website status
def error(request, template_name="500.html"):
    FEC_CMS_ENVIRONMENT = request.GET.get("FEC_CMS_ENVIRONMENT", "")
    if settings.FEATURES.get('website_status'):
    ### return server_error(request, '500-status.html', {'FEC_CMS_ENVIRONMENT' : 
FEC_CMS_ENVIRONMENT })
        return render(
        request,
        '500-status.html',
        {'FEC_CMS_ENVIRONMENT' : FEC_CMS_ENVIRONMENT })
### return server_error(request, template_name)
    return render(
        request,
        template_name,
        {'FEC_CMS_ENVIRONMENT' : FEC_CMS_ENVIRONMENT })
```

**Comment by patphongs** [comment]
> @patphongs Related to an issue you pointed out with the environment not showing on the env-specific banners for 500 errors: #3708.
> We could pass it into this view. But this could also just be a done for the followup issue to the env-banners.
> 
> ```
> # Custom handle 500 error for website status
> def error(request, template_name="500.html"):
>     FEC_CMS_ENVIRONMENT = request.GET.get("FEC_CMS_ENVIRONMENT", "")
>     if settings.FEATURES.get('website_status'):
>     ### return server_error(request, '500-status.html')
>         return render(
>         request,
>         '500-status.html',
>         {'FEC_CMS_ENVIRONMENT' : FEC_CMS_ENVIRONMENT })
> ### return server_error(request, template_name)
>     return render(
>         request,
>         template_name,
>         {'FEC_CMS_ENVIRONMENT' : FEC_CMS_ENVIRONMENT })
> ```

Great suggestion @johnnyporkchops! I have incorporated the environment in the 500 error message view

File: fec/home/views.py [hunk 1]
```diff
@@ -399,3 +401,17 @@ def guides(request):
         "home/candidate-and-committee-services/guides.html",
         {"self": page_context},
     )
+
+# Custom handle 500 error for website status
+def error(request, template_name="500.html"):
+    FEC_CMS_ENVIRONMENT = request.GET.get("FEC_CMS_ENVIRONMENT", "")
```
Comment (johnnyporkchops) [inline fec/home/views.py:407]: I was thinking may FEC_CMS_ENVIRONMENT var does not have to be uppercase here, and should be l/c for consistency with other views.

Comment (patphongs) [reply fec/home/views.py]: Thanks, updated to `cms_env`

File: fec/fec/templates/partials/env-banner.html [hunk 2]
```diff
@@ -1,16 +1,18 @@
 {% if settings  != undefined %}
     {% if settings.FEC_CMS_ENVIRONMENT != 'PRODUCTION' %}   
       <div class="banner ">
+        {# Uses settings.FEC_CMS_ENVIRONMENT for HTML templates #}
         <p class="t-sans t-small">{{settings.FEC_CMS_ENVIRONMENT}}<br>
-        This site is for testing ideas and code. View the most accurate data on the <a href="http://www.fec.gov">live site</a>.
+        HTML This site is for testing ideas and code. View the most accurate data on the <a href="http://www.fec.gov">live site</a>.
```
Comment (lbeaufort) [inline fec/fec/templates/partials/env-banner.html:6]: @patphongs I think we need to take out these `HTML` and `Jinja` labels

File: fec/home/views.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from datetime import datetime
 from itertools import chain
 from operator import attrgetter
+from django.http import HttpResponse
```
Comment (lbeaufort) [inline fec/home/views.py:4]: Is this a double-import? See line 7

Comment (patphongs) [reply fec/home/views.py]: Thanks! Removed the extra import

File: fec/fec/templates/partials/env-banner.html (hunk 2 above)
Comment (patphongs) [reply fec/fec/templates/partials/env-banner.html]: Thanks! Removed those from the banners

**Pull request merged**